jf_agent/git/bitbucket_cloud_adapter.py

```diff
@@ -142,35 +142,44 @@ def get_repos(
 
     @diagnostics.capture_timing()
     @agent_logging.log_entry_exit(logger)
-    def get_default_branch_commits(
-        self, normalized_repos: List[NormalizedRepository], server_git_instance_info,
+    def get_commits_for_included_branches(
+        self, normalized_repos: List[NormalizedRepository], included_branches: dict, server_git_instance_info,
     ) -> List[NormalizedCommit]:
-        print('downloading bitbucket default branch commits... ', end='', flush=True)
+        print('downloading bitbucket commits on included branches... ', end='', flush=True)
+
         for i, repo in enumerate(normalized_repos, start=1):
             with agent_logging.log_loop_iters(logger, 'repo for branch commits', i, 1):
                 pull_since = pull_since_date_for_repo(
                     server_git_instance_info, repo.project.login, repo.id, 'commits'
                 )
-                for j, commit in enumerate(
-                    tqdm(
-                        self.client.get_commits(repo.project.id, repo.id, repo.default_branch_name),
-                        desc=f'downloading commits for {repo.name}',
-                        unit='commits',
-                    ),
-                    start=1,
-                ):
-                    with agent_logging.log_loop_iters(logger, 'branch commit inside repo', j, 100):
-                        commit = _normalize_commit(
-                            commit,
-                            repo,
-                            self.config.git_strip_text_content,
-                            self.config.git_redact_names_and_urls,
-                        )
-                        yield commit
 
-                        # yield one commit older than we want to see
-                        if commit.commit_date < pull_since:
-                            break
+                # Find branches for which we should pull commits, specified by customer in config.
```

Comment:
There is some existing (and newly) duplicate code across the 4 implementations of git that we touch. I created a story to address this and have all implementations use the Adapter class that already exists. Hopefully that will let us reuse more code.
https://jelly-ai.atlassian.net/browse/OJ-11355